Add interview page (UI)

AndresNeisa opened the pull request:
The interview page makes use of the Question, CodeEditor, and Console components. I am attaching images for 3 different screen sizes.

### Big desktop screen:
![Interview UI](https://user-images.githubusercontent.com/67347417/115574771-ee7b1980-a28f-11eb-9f69-7d6dbd223139.JPG)

### Small desktop screen:
![localhost_3000_interview_456](https://user-images.githubusercontent.com/67347417/115576057-23d43700-a291-11eb-98e9-012b0b96e2d2.png)

### Mobile screen:
![localhost_3000_interview_456 (1)](https://user-images.githubusercontent.com/67347417/115576033-1d45bf80-a291-11eb-9c2f-f120fdb5417c.png)

AndresNeisa committed:
Add the interview page with a basic code editor

AndresNeisa committed:
Add question view into Interview page

AndresNeisa committed:
Add question preview in the interview page

AndresNeisa committed:
Remove dialog from the interview page

The content of the interview was inside a dialog which has been removed.
The removal of the dialog caused a few bugs that have been solved as well.

AndresNeisa committed:
Add the console to the Interview page

AndresNeisa committed:
Fix layout in smaller devices

Some miner modifications to the media queries have been added. The border of the buttons on the interview page have also been added.

AndresNeisa committed:
Merge branch 'dev' into code-dialog

dproc96 commented on client/src/components/UpcomingInterviews.js:
```diff
@@ -90,6 +98,11 @@ export default function UpcomingInterviews( { rows } ) {
                   <StyledTableCell>{row.theme}</StyledTableCell>
                     <StyledTableCell align="right">
+                      {row.live && 
+                        <Link className={classes.link} to={`interview/${row.id}`}>
+                        <Button variant='outlined' className={classes.btn}>Go to interview</Button>
+                        </Link>
```
Can you fix the indentation here?

dproc96 commented on client/src/components/layout/Question.js:
```diff
@@ -0,0 +1,93 @@
+
+import ReactMarkdown from 'react-markdown';
+import {Prism as SyntaxHighlighter} from 'react-syntax-highlighter';
+import gfm from 'remark-gfm';
```
This import is failing because it's not in the package.json

AndresNeisa replied:
The missing packages have been already added to the package.json file

dproc96 commented on client/src/components/layout/Console.js:
```diff
@@ -0,0 +1,84 @@
+                </Toolbar>
+            </AppBar>
+            <ControlledEditor 
+            value={value}
+            className={classes.codeMirrorWrapper}
+            options={{
+                lineWrapping: true,
+                lint: true,
+                theme: 'material',
+                tabSize: 2,
+                fixedGutter: true,
+                scrollbarStyle: 'null'
```
Can you fix the spacing here

dproc96 commented on client/src/components/layout/Question.js:
```diff
@@ -0,0 +1,93 @@
+
+const components = {
+  code({node, inline, className, children, ...props}) {
+    const match = /language-(\w+)/.exec(className || '')
```
Can you run Prettier on your files, should fix all the code styling issues and missing semicolons

AndresNeisa replied:
Sure, I will see what is going on with the Prettier extension as it is not formatting my files properly.

dproc96 commented on client/src/pages/TempHome.js:
```diff
@@ -1,24 +1,34 @@
 
 
 const Home = () => {
+  const [navbarHieght, setHeightnavbarHieght] = useState(0);
```
```suggestion
  const [navbarHeight, setNavbarHeight] = useState(0);
```

AndresNeisa replied:
Typo has already been fixed

dproc96 reviewed (commented):
Had a few comments, UI looks good in the screenshots though!

AndresNeisa committed:
Update dashboard.js

AndresNeisa committed:
Merge branch 'dev' into code-dialog

AndresNeisa committed:
Fix comments made in the review

Prettier wasn't working on my side, I solve the issue and format the files.
The missing packages were included in package.json.
Height typo has been fixed

AndresNeisa committed:
Update Dashboard.js

AndresNeisa committed:
Remove dashboard files

AndresNeisa committed:
Create Dashboard.js

AndresNeisa committed:
Delete Dashboard.js

AndresNeisa committed:
Create Dashboard.js

AndresNeisa committed:
Merge branch 'dev' into code-dialog

dproc96 reviewed (commented):
You may want to check your Prettier settings with your group, it looks like it went through and changed a bunch of formatting

jessdesr reviewed (approved):
Prettier can be fixed in a followup commit, I'm okay with this being merged now 👍
